Refactor the tests

nc6 opened the pull request:
Addresses #1681 

This splits the test more thoroughly into lib and test sections, and avoids confusion between the two. As a result, we drop some dependencies on parts of examples in generators. 

- Move Control.Iterate.SetAlgebra tests into a new, more sensible location
- Split the bootstrap generators from tests
- Serialisation generators no longer rely on the `NonTraceProperties` generators.

nc6 committed:
Sensible shelley-spec-ledger-test library

Remove the dependency on the hedgehog generators from the serialisation
generators.

nc6 committed:
Fix up test suite.

JaredCorduan reviewed (approved):
LGTM! I'm happy to have less confusion between the two cabal files!

uroboros reviewed (approved):
👍

nc6 committed:
Fixup generators.

Commit:
fix Script deserialization bug

and also contain arbitrary instances for the two recursize types, metadatum and multisig scripts

nc6 committed:
Nix fixes. Nixes.

nc6 committed:
Rebase fixes
